pkg/services/query/errors.go

```diff
@@ -1,17 +1,19 @@
 package query
 
-import "fmt"
+import (
+	"errors"
 
-// ErrBadQuery returned whenever request is malformed and must contain a message
-// suitable to return in API response.
-type ErrBadQuery struct {
-	Message string
-}
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/util/errutil"
+)
 
-func NewErrBadQuery(msg string) *ErrBadQuery {
-	return &ErrBadQuery{Message: msg}
-}
+var ErrBadQuery = errutil.NewBase(errutil.StatusBadRequest, "query.badRequest").MustTemplate("Bad plugin request: {{ .Error }}", errutil.WithPublic("{{ .Public.Message }}"))
```

Comment:
Think we should rename this to `ErrQueryValidationFailure` or similar with error `query validation failed: 

```suggestion
var ErrBadQuery = errutil.NewBase(errutil.StatusBadRequest, "query.badRequest").MustTemplate("bad plugin request: {{ .Error }}", errutil.WithPublic("{{ .Public.Message }}"))
```